app/models/section.rb

```diff
@@ -181,10 +188,14 @@ def all_editions
 
 private
 
-  attr_reader :slug_generator, :editions, :latest_edition
+  attr_reader :slug_generator, :latest_edition, :previous_edition
 
   def published_edition
-    most_recent_non_draft = editions.reject(&:draft?).last
+    most_recent_non_draft = if !latest_edition.draft?
+                              latest_edition
+                            elsif previous_edition && !previous_edition.draft?
+                              previous_edition
+                            end
```

Comment:
Good spot. I had it in mind that I was changing existing code when I addressed the Rubocop feedback but I was actually only changing the code that I'd just added. I've removed my comment about Rubocop from the commit note.